gammapy/datasets/evaluator.py

```diff
@@ -366,7 +375,13 @@ def apply_edisp(self, npred):
         npred_reco : `~gammapy.maps.Map`
             Predicted counts in reconstructed energy bins.
         """
-        return apply_edisp(npred, self.edisp)
+        if self.model.apply_irf["edisp"]:
+            return apply_edisp(npred, self.edisp)
+        else:
+            if "energy_true" in npred.geom.axes.names:
+                return apply_edisp(npred, self._edisp_diagonal)
+            else:
+                return npred
```

Comment:
This was the previous behaviour, no? Strange it is not covered by the tests

Reply:
the previous behaviour was equivalent to return npred even if "energy_true" in npred.geom.axes.names, which lead to a geom inconsistency